### ISSUE TITLE
Sorting for list endpoints

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-api#555]
Discussion started here: https://github.com/Open-EO/openeo-api/issues/550#issuecomment-2548217977

I think we could directly add parameters to the API, we already have it implicitly for the STAC part (through the sort extension) and could simply copy that to the other endpoints. That should be low effort (on the specification side).

We'd probably not define default values (i.e. if not provided it's up to the implementation), but we can recommend defaults.

We have pagination defined for the following endpoints and the following recommendations probably make sense:
- `/collections`: `id`
- `/processes`: `id`
- `/process_graphs`: `id`
- `/services`: `created` (property not required though => nevertheless: creation sequence/time)
- `/jobs` `created`
- `/files` `path`
- logs (services and jobs): `time` (property not required though => nevertheless: logging sequence/time)
- `/workspaces`: not sure (creation? but there's no corresponding property for it)
- `/orders`: `order:date`

By the way, pagination in the Web Editor is pretty complex and leads to various complexities and potentially confusing behaviour the way it is implemented right now. The "refresh" button should probably be removed.

> leave it to clients to fill in the UX gaps (e.g. show alphabetical even if pagination is not like that)

Whenever I tried that, it felt really weird. Especially, if sorted alphabetically, it's completely confusing if you paginate and start from A again on every page.

**Comment by soxofaan** [comment]
> >    leave it to clients to fill in the UX gaps (e.g. show alphabetical even if pagination is not like that)
>
> Whenever I tried that, it felt really weird. Especially, if sorted alphabetically, it's completely confusing if you paginate and start from A again on every page.

Indeed, mixing different keys for pagination and sorting should not be the default, but at most a possibility for the user if they really want and are aware of the implications
